src/core/inventory.c

```
@@ -419,7 +419,7 @@ provides_cb(mender_keystore_t **inventory, uint8_t *inventory_len) {
        structure for manipulation with provides data in other places), but we
        need a key-value array so we need to do a transformation. */
     if (MENDER_OK != (ret = mender_storage_get_provides(&provides))) {
-        mender_log_error("Failed to get provides from storage");
+        /* error logged in function */
```

We should only hide the error when the error is `MENDER_NOT_FOUND`.

---

That was how I initially did it, but then I saw that we log `Unable to read provides` in the function; is there any value in logging `Failed to get provides from storage` as well?

---

Right, then I would expect `Unable to read provides` to be printed as a warning or even info message. It is a totally valid case not to have provides, a device can live and get updates without provides and depends functionality from the Artifacts.

I think this is what you addressed now on your second commit. I'll have a look :eyes: